src/main/java/edu/wpi/grip/core/Source.java

```diff
@@ -45,4 +45,44 @@
      * @see #getProperties()
      */
     public abstract void createFromProperties(EventBus eventBus, Properties properties) throws IOException;
+
+
+    public <T extends Source> T start(EventBus eventBus) throws IOException {
+        final T source = (T) start();
+        eventBus.register(source);
+        return source;
+    }
+
+    /**
+     * Starts this source.
+     * A source whose {@link #isRestartable()} returns true can also be stopped and started and stopped multiple times.
+     *
+     * @return The source object that created the camera
+     * @throws IOException If the source fails to be started
+     */
+    protected abstract Source start() throws IOException;
+
+    /**
+     * Stops this source.
+     * This will stop the source publishing new socket values after this method returns.
+     * A source whose {@link #isRestartable()} returns true can also be stopped and started and stopped multiple times.
+     *
+     * @return The source that was stopped
+     * @throws Exception
+     */
+    public abstract Source stop() throws Exception;
```

Comment:
I guess this could be changed to an IO Exception as well because we are dealing with Inputs and Outputs